packages/contracts-periphery/contracts/L1/messaging/L1ERC721Bridge.sol

```diff
@@ -1,168 +1,244 @@
+     *
+     * @param localToken  Address of the token on this domain.
+     * @param remoteToken Address of the token on the remote domain.
+     * @param from        Address that initiated bridging action.
+     * @param to          Address to receive the token.
+     * @param tokenId     ID of the specific token deposited.
+     * @param data        Extra data for use on the client-side.
+     */
+    event ERC721BridgeFinalized(
+        address indexed localToken,
+        address indexed remoteToken,
+        address indexed from,
+        address to,
+        uint256 tokenId,
+        bytes data
+    );
+
+    /**
+     * @notice Address of the bridge on the other network.
+     */
+    address public otherBridge;
 
     // Maps L1 token to L2 token to token ID to a boolean indicating if the token is deposited
+    /**
+     * @notice Mapping of L1 token to L2 token to ID to boolean, indicating if the given L1 token
+     *         by ID was deposited for a given L2 token.
+     */
     mapping(address => mapping(address => mapping(uint256 => bool))) public deposits;
 
-    /***************
-     * Constructor *
-     ***************/
-
-    // This contract lives behind a proxy, so the constructor parameters will go unused.
-    constructor(address _l1messenger, address _l2ERC721Bridge) CrossDomainEnabled(address(0)) {
-        _initialize(_l1messenger, _l2ERC721Bridge);
-    }
-
-    /******************
-     * Initialization *
-     ******************/
-
     /**
-     * @param _l1messenger L1 Messenger address being used for cross-chain communications.
-     * @param _l2ERC721Bridge L2 ERC721 bridge address.
+     * @param _messenger   Address of the CrossDomainMessenger on this network.
+     * @param _otherBridge Address of the ERC721 bridge on the other network.
      */
-    function _initialize(address _l1messenger, address _l2ERC721Bridge) internal {
-        messenger = _l1messenger;
-        l2ERC721Bridge = _l2ERC721Bridge;
+    constructor(address _messenger, address _otherBridge) CrossDomainEnabled(address(0)) {
+        initialize(_messenger, _otherBridge);
     }
 
-    /**************
-     * Depositing *
-     **************/
-
     /**
-     * @inheritdoc IL1ERC721Bridge
+     * @param _messenger   Address of the CrossDomainMessenger on this network.
+     * @param _otherBridge Address of the ERC721 bridge on the other network.
      */
-    function depositERC721(
-        address _l1Token,
-        address _l2Token,
-        uint256 _tokenId,
-        uint32 _l2Gas,
-        bytes calldata _data
-    ) external virtual {
-        // Modifier requiring sender to be EOA.  This check could be bypassed by a malicious
-        // contract via initcode, but it takes care of the user error we want to avoid.
-        require(!Address.isContract(msg.sender), "Account not EOA");
+    function initialize(address _messenger, address _otherBridge) public initializer {
```

Comment:
since this is `initializer`, it will fail to be called a second time after the constructor